executor/read/pattern_executor.rs

```diff
@@ -157,12 +145,14 @@ impl PatternExecutor {
                     else {
                         unreachable!();
                     };
-                    let result = inner.compute_next_batch_with_retries(
+                    let mut negation_suspensions = QueryPatternSuspensions::new_root();
```

Comment:
We basically force the Negation to finish with this right? Ie. all suspensions have to be resolved?

Reply:
Yes. The `root` suspensions should never have any points added to them.